### DISCUSSION TITLE
Will wikijs 3.0 see the ligth?

### DISCUSSION THREAD

**Discussion by bobonov** [requarks/wiki#7011] [Help / Questions]
I follow and use wikijs with interest since 3 years now and 3.0 version has been announced in 2021 if not before.
On October 30, 2022, a Developer Preview was released and "A beta will be available in the coming weeks / months, once more features are stable and working.", no beta has been released till today.
That means that not all 2.x feature have been ported in 3.0, while at same time new feature are added and planned.
It is my impression that more time pass and more intriguing new technology came up and more features are added to 3.0.
During these years I saw more feature added than closed one and I really start to think that this is going to be an endless work.
By now there should be a roadmap with planned date or at least with planned year’s quarter for the release.
All new feature not ready to be released should be frozen and postponed and all 2.x needed feature not ported should get priority.

**Comment by fnafox** [comment]
This is an open source project, if you want to backport 3.0 features nobody is stopping you from doing so and submitting PRs.
There's an issue featuring progress here: https://github.com/requarks/wiki/issues/6844

**Reply by NormskiJJO** [reply]
Never

**Comment by bobonov** [comment]
I do not speak about backporting, but about planning. If you do not make a plan and freeze the new feature but keep adding, it became an endless work. Every 2 month there is something new that can be added.
Maybe a solution could be by making wikijs modular, so its core has only basic functionality (like auth only via db) and other extra can be downloaded via a module repository so new feature can be implemented at any stage.
But I think it is too late to make such change.

**Reply by NGPixel** [reply]
No new features are constantly being added. The goal is simply to have feature parity with v2 as listed in #6844.
Wiki.js is already modular (has been since day 1) and it's already the case that only the local auth is working.

**Reply by bobonov** [reply]
But modularity without the possibility to release only the module and add it to the software later on using a module repository  without a complete upgrade is almost useless. This is the difference between monolithic and modularity. If you implement modularity, but you are still stuck to a complete upgrade is monolithic anyway. In wikijs the modularity is only partial, on the software side but not in the concept.

I know about the feature parity but still new feature were added. According to the news on the site, newly implemented feature are:

**January 8, 2021**
Page Relations
**March 21, 2021**
Multi-Sites
**April 24, 2021**
Editors: you can now select which editor is available to your users;
Security: A new CORS option has been added
**May 2021**
Here's some new features and improvements coming in the future 3.0 release.
Login Experience: rewrite due to multisite
Profile: The profile page has been completely redesigned along with some new sections:
•	Manage notifications (more details in a future post)
•	View activity (such as logins, page edits, etc.)
Live Collaboration
**July 2021**
Webhooks
Extensions
**September 7, 2021**
Semantic Data
**November 7, 2021**
Managing Users, Metadata: a new Metadata tab has been added with a full JSON editor. While **they won't have any use at first**, the plan is to allow integrations to query these fields and possibly use them inside pages as well. **(so a new feature that to be used need further implementation)**
**December 2021**
There are some major improvements coming to storage in Wiki.js 3.0. First off, is the ability to have multiple storage targets per site. It's also now possible to define which type of content should be synchronized / backed up to the storage target. …… a new GitHub storage module….
While advertise as an improvement these are new features
**January 28, 2022**
GitHub Setup (due to the above new GitHub storage module)
**February 6, 2022**
Feature Preview - Storage Delivery Paths
As outlined in the Wiki.js 3.0 Feature Preview - Storage post, you'll be able to store and serve assets from different storage targets based on their type and size.
In order the visualize how that content is delivered to the user, Wiki.js 3.0 introduces a new view called "Delivery Paths".
**May 1, 2022**
The Road to the Wiki.js 3.0 Beta: In order to get Wiki.js 3.0 in a state that can be used and tested by the community as soon as possible, here is the roadmap for the coming months (maybe it was meant years since no beta came out till today)
**September 9, 2022**
Admin Terminal: Wiki.js 3.0 introduces a new Terminal page
**January 22, 2023**
File Manager: Wiki.js 3.0 introduces a new management system called File Manager.
**September 7, 2023**
Blocks: Native support for Web Components is been added in Wiki.js 3.0. They are called blocks…..
**October 12, 2023**
Passkeys: Passkeys are coming to Wiki.js 3.0.

**Reply by NGPixel** [reply]
That's not how it works. Wiki.js is modular for all the things that **CAN** be modular. For example, you can't make "multi-sites" modular... it's a core functionality of the system. Almost all modules that were developed for v2 will work in v3.

Blog posts about new features don't always represent what's actually being shipped. Many posts are experimentation or UI concepts (e.g. semantic data, live collaboration, notifications, etc.).

The goal of v3 is to provide a foundation for many requested features that were simply not possible to implement in v2 directly.

The feature parity post was posted **after** all the "new features" you listed. You make it sound like it was announced from the beginning and that new features were still being added afterwards, which is completely not true.

**Reply by bobonov** [reply]
This list it was not to say that all of them can be module. It is only the list of all the main new feature announced for 3.0 version. Anyway, again if the module **can't be installed independently but require a full upgrade, the application is still monolithic in the usage**.
Beside the fact that porting all the functionality to the new version (unless deprecated) it is something obvious, and for me, it is implied in the first announcement of the 3.0
Looking at the news:

- In May 2022 you announced a roadmap to the beta
- In October 2022 A beta will be available in the coming weeks / months
- Beta Parity feature is the roadmap.

But new functionalities have been added afterward.
Anyway, I do not want to make a flame discussion and have the developers lose their time. I just needed to understand if there is some clear path.
Good luck.

**Reply by SgtErnestBilko** [reply]
>Anyway, I do not want to make a flame discussion and have the developers lose their time. I just needed to understand if there is some clear path.
Good luck.

I too appreciate all the hard work going into this project.  I am not a developer so am unable to contribute.  I would be  happy to contribute financially when v3.0 is released.  It has features I need.   Are you days, weeks, months or years away from releasing v3?  I truly am not flaming the developers and reiterate that I appreciate all your hard work.

**Comment by TitaniumHocker** [comment]
![image](https://github.com/requarks/wiki/assets/37049574/8e288104-73ac-4488-9c8a-3c79302b45ab)

**Comment by gavenwan** [comment]
it's a Wonderfull project , look forward to 3.0
